pkg/alertmanager/operator.go

```diff
@@ -162,6 +156,49 @@ func (c *Operator) RegisterMetrics(r prometheus.Registerer) {
 	)
 }
 
+// waitForCacheSync waits for the informers' caches to be synced.
+func (c *Operator) waitForCacheSync(stopc <-chan struct{}) error {
+	ok := true
+	type informerSynced struct {
+		name string
+		sync cache.InformerSynced
+	}
+	informers := map[*informerSynced]bool{
```

Comment:
The map cannot use an anonymous struct (or even a named struct type) as its key because the `sync` field is not comparable. Even if the field was `informer cache.SharedInformer` then we'd be back at the original issue Sergiusz raised where we'd only know if the field was comparable at runtime. Instead, we can use a pointer, whose address is comparable.

Reply:
Suggestion: let's omit the map and go with a simpler slice:
```
	type namedInformer struct {
		name     string
		informer cache.SharedIndexInformer
	}

	informers = append(informers, namedInformer{"Alertmanager", c.alrtInf})
	informers = append(informers, namedInformer{"StatefulSet", c.alrtInf})

	for _, inf := range informers {
		if !cache.WaitForCacheSync(stopc, inf.informer.HasSynced) {
			level.Error(c.logger).Log("msg", fmt.Sprintf("failed to sync %s cache", inf.name))
			ok = false
		} else {
			level.Debug(c.logger).Log("msg", fmt.Sprintf("successfully synced %s cache", inf.name))
		}
	}
```